src/functions.rs

```diff
@@ -289,4 +311,22 @@ mod tests {
         assert!(!access_syscall("src/lib.rs", X_OK).unwrap());
         assert!(!is_removable("/").unwrap());
     }
+
+    #[test]
+    #[cfg(windows)]
+    fn test_access_syscall() {
+        assert!(access_syscall("src/", F_OK | R_OK | W_OK).unwrap());
+        assert!(access_syscall("src/", F_OK).unwrap());
+        assert!(access_syscall("src/", R_OK).unwrap());
+        assert!(access_syscall("src/", W_OK).unwrap());
+
+        assert!(access_syscall("src/lib.rs", R_OK).unwrap());
+        assert!(access_syscall("src/lib.rs", W_OK).unwrap());
+        assert!(access_syscall("src/lib.rs", F_OK).unwrap());
+
+        assert!(access_syscall("path_doesnt_exist/", F_OK).is_err()); // invalid path
+        // assert!(access_syscall("src/", 0b11111).is_err()); // invalid number
```

Comment:
Should this be commented out?

Reply:
I completely forgot about this. This crashed the windows tests with a stackoverflow. I don't know why, but the winapi function can't handle invalid bytes.

I added the sanity check myself and return the proper error according to the winapi docs.